openingd/dualopend.c

```diff
@@ -3185,20 +3241,9 @@ static void opener_start(struct state *state, u8 *msg)
 
 	/* We need to check that the inputs we've already provided
 	 * via the API are confirmed :/ */
-	if (state->require_confirmed_inputs) {
-		msg = towire_dualopend_validate_inputs(NULL, tx_state->psbt,
-						       state->our_role);
-		wire_sync_write(REQ_FD, take(msg));
-		msg = wire_sync_read(tmpctx, REQ_FD);
-
-		if (!fromwire_dualopend_validate_inputs_reply(msg)) {
-			if (!fromwire_dualopend_fail(msg, msg, &err_reason))
-				master_badmsg(fromwire_peektype(msg), msg);
-			/* We abort, because we don't have valid inputs */
-			open_abort(state, "%s", err_reason);
-			return;
-		}
-	}
+	if (state->require_confirmed_inputs[REMOTE] &&
+	    !validate_inputs(state, tx_state, state->our_role))
+		return;
```

Comment:
This looks a bit weird, right?  Maybe get validate_inputs() to return NULL or errmsg, and then do the open_abort() here if return is non-NULL?  Fits the pattern in the rest, and seems more explicit?